File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/ViewUpdateDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The details for updating a view.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ViewUpdateDetails.Builder.class)

public final class ViewUpdateDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"displayName", "description", "viewText", "viewFields", "addViewProperties", "dropViewProperties"})
    public ViewUpdateDetails(String displayName, String description, String viewText, java.util.List<ViewFieldDetails> viewFields, java.util.List<ViewProperty> addViewProperties, java.util.List<ViewProperty> dropViewProperties) {
        super();
        this.displayName = displayName;
        this.description = description;
        this.viewText = viewText;
        this.viewFields = viewFields;
        this.addViewProperties = addViewProperties;
        this.dropViewProperties = dropViewProperties;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

                /**
         * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
         * @param displayName the value to set
         * @return this builder
         **/
        

        public Builder displayName(String displayName) {
        this.displayName = displayName;
        return this;
        }
            /**
     * The description of the view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

                /**
         * The description of the view.
         * @param description the value to set
         * @return this builder
         **/
        

        public Builder description(String description) {
        this.description = description;
        return this;
        }
            /**
     * The Query used to create the view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("viewText")
        private String viewText;

                /**
         * The Query used to create the view.
         * @param viewText the value to set
         * @return this builder
         **/
        

        public Builder viewText(String viewText) {
        this.viewText = viewText;
        return this;
        }
            /**
     * Columns for view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("viewFields")
        private java.util.List<ViewFieldDetails> viewFields;

                /**
         * Columns for view.
         * @param viewFields the value to set
         * @return this builder
         **/
        

        public Builder viewFields(java.util.List<ViewFieldDetails> viewFields) {
        this.viewFields = viewFields;
        return this;
        }
            /**
     * View Properties.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("addViewProperties")
        private java.util.List<ViewProperty> addViewProperties;

                /**
         * View Properties.
         * @param addViewProperties the value to set
         * @return this builder
         **/
        

        public Builder addViewProperties(java.util.List<ViewProperty> addViewProperties) {
        this.addViewProperties = addViewProperties;
        return this;
        }
            /**
     * View Properties.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("dropViewProperties")
        private java.util.List<ViewProperty> dropViewProperties;

                /**
         * View Properties.
         * @param dropViewProperties the value to set
         * @return this builder
         **/
        

        public Builder dropViewProperties(java.util.List<ViewProperty> dropViewProperties) {
        this.dropViewProperties = dropViewProperties;
        return this;
        }


        public ViewUpdateDetails build() {
            ViewUpdateDetails model = new ViewUpdateDetails(this.displayName
                    , this.description
                    , this.viewText
                    , this.viewFields
                    , this.addViewProperties
                    , this.dropViewProperties);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ViewUpdateDetails model) {
                this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.viewText(model.getViewText());
    this.viewFields(model.getViewFields());
    this.addViewProperties(model.getAddViewProperties());
    this.dropViewProperties(model.getDropViewProperties());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * A user-friendly name. Has to be unique within the scope of the schema and is changeable.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * The description of the view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * The description of the view.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * The Query used to create the view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("viewText")
    private final String viewText;

        /**
     * The Query used to create the view.
     * @return the value
     **/
    
    public String getViewText() {
        return viewText;
    }


        /**
     * Columns for view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("viewFields")
    private final java.util.List<ViewFieldDetails> viewFields;

        /**
     * Columns for view.
     * @return the value
     **/
    
    public java.util.List<ViewFieldDetails> getViewFields() {
        return viewFields;
    }


        /**
     * View Properties.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("addViewProperties")
    private final java.util.List<ViewProperty> addViewProperties;

        /**
     * View Properties.
     * @return the value
     **/
    
    public java.util.List<ViewProperty> getAddViewProperties() {
        return addViewProperties;
    }


        /**
     * View Properties.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("dropViewProperties")
    private final java.util.List<ViewProperty> dropViewProperties;

        /**
     * View Properties.
     * @return the value
     **/
    
    public java.util.List<ViewProperty> getDropViewProperties() {
        return dropViewProperties;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ViewUpdateDetails(");
        sb.append("displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", viewText=").append(String.valueOf(this.viewText));
        sb.append(", viewFields=").append(String.valueOf(this.viewFields));
        sb.append(", addViewProperties=").append(String.valueOf(this.addViewProperties));
        sb.append(", dropViewProperties=").append(String.valueOf(this.dropViewProperties));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ViewUpdateDetails)) {
            return false;
        }

        ViewUpdateDetails other = (ViewUpdateDetails) o;
        return java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.viewText, other.viewText) &&
            java.util.Objects.equals(this.viewFields, other.viewFields) &&
            java.util.Objects.equals(this.addViewProperties, other.addViewProperties) &&
            java.util.Objects.equals(this.dropViewProperties, other.dropViewProperties);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.viewText == null ? 43 : this.viewText.hashCode());
        result = (result * PRIME) + (this.viewFields == null ? 43 : this.viewFields.hashCode());
        result = (result * PRIME) + (this.addViewProperties == null ? 43 : this.addViewProperties.hashCode());
        result = (result * PRIME) + (this.dropViewProperties == null ? 43 : this.dropViewProperties.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/view_update_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ViewUpdateDetails(object):
    """
    The details for updating a view.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ViewUpdateDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param display_name:
            The value to assign to the display_name property of this ViewUpdateDetails.
        :type display_name: str

        :param description:
            The value to assign to the description property of this ViewUpdateDetails.
        :type description: str

        :param view_text:
            The value to assign to the view_text property of this ViewUpdateDetails.
        :type view_text: str

        :param view_fields:
            The value to assign to the view_fields property of this ViewUpdateDetails.
        :type view_fields: list[oci.aidataplatform_dp.models.ViewFieldDetails]

        :param add_view_properties:
            The value to assign to the add_view_properties property of this ViewUpdateDetails.
        :type add_view_properties: list[oci.aidataplatform_dp.models.ViewProperty]

        :param drop_view_properties:
            The value to assign to the drop_view_properties property of this ViewUpdateDetails.
        :type drop_view_properties: list[oci.aidataplatform_dp.models.ViewProperty]

        """
        self.swagger_types = {
            'display_name': 'str',
            'description': 'str',
            'view_text': 'str',
            'view_fields': 'list[ViewFieldDetails]',
            'add_view_properties': 'list[ViewProperty]',
            'drop_view_properties': 'list[ViewProperty]'
        }

        self.attribute_map = {
            'display_name': 'displayName',
            'description': 'description',
            'view_text': 'viewText',
            'view_fields': 'viewFields',
            'add_view_properties': 'addViewProperties',
            'drop_view_properties': 'dropViewProperties'
        }

        self._display_name = None
        self._description = None
        self._view_text = None
        self._view_fields = None
        self._add_view_properties = None
        self._drop_view_properties = None

    @property
    def display_name(self):
        """
        Gets the display_name of this ViewUpdateDetails.
        A user-friendly name. Has to be unique within the scope of the schema and is changeable.


        :return: The display_name of this ViewUpdateDetails.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this ViewUpdateDetails.
        A user-friendly name. Has to be unique within the scope of the schema and is changeable.


        :param display_name: The display_name of this ViewUpdateDetails.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        Gets the description of this ViewUpdateDetails.
        The description of the view.


        :return: The description of this ViewUpdateDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this ViewUpdateDetails.
        The description of the view.


        :param description: The description of this ViewUpdateDetails.
        :type: str
        """
        self._description = description

    @property
    def view_text(self):
        """
        Gets the view_text of this ViewUpdateDetails.
        The Query used to create the view.


        :return: The view_text of this ViewUpdateDetails.
        :rtype: str
        """
        return self._view_text

    @view_text.setter
    def view_text(self, view_text):
        """
        Sets the view_text of this ViewUpdateDetails.
        The Query used to create the view.


        :param view_text: The view_text of this ViewUpdateDetails.
        :type: str
        """
        self._view_text = view_text

    @property
    def view_fields(self):
        """
        Gets the view_fields of this ViewUpdateDetails.
        Columns for view.


        :return: The view_fields of this ViewUpdateDetails.
        :rtype: list[oci.aidataplatform_dp.models.ViewFieldDetails]
        """
        return self._view_fields

    @view_fields.setter
    def view_fields(self, view_fields):
        """
        Sets the view_fields of this ViewUpdateDetails.
        Columns for view.


        :param view_fields: The view_fields of this ViewUpdateDetails.
        :type: list[oci.aidataplatform_dp.models.ViewFieldDetails]
        """
        self._view_fields = view_fields

    @property
    def add_view_properties(self):
        """
        Gets the add_view_properties of this ViewUpdateDetails.
        View Properties.


        :return: The add_view_properties of this ViewUpdateDetails.
        :rtype: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        return self._add_view_properties

    @add_view_properties.setter
    def add_view_properties(self, add_view_properties):
        """
        Sets the add_view_properties of this ViewUpdateDetails.
        View Properties.


        :param add_view_properties: The add_view_properties of this ViewUpdateDetails.
        :type: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        self._add_view_properties = add_view_properties

    @property
    def drop_view_properties(self):
        """
        Gets the drop_view_properties of this ViewUpdateDetails.
        View Properties.


        :return: The drop_view_properties of this ViewUpdateDetails.
        :rtype: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        return self._drop_view_properties

    @drop_view_properties.setter
    def drop_view_properties(self, drop_view_properties):
        """
        Sets the drop_view_properties of this ViewUpdateDetails.
        View Properties.


        :param drop_view_properties: The drop_view_properties of this ViewUpdateDetails.
        :type: list[oci.aidataplatform_dp.models.ViewProperty]
        """
        self._drop_view_properties = drop_view_properties

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
